js/main.js

```diff
@@ -1,2 +1,41 @@
 /* global data */
 /* exported data */
+
+var urlInput = document.querySelector('#input-url');
+var photo = document.querySelector('#main-image');
+
+function updatePhoto(event) {
+  photo.setAttribute('src', event.target.value);
+  if (event.target.value === '') {
+    photo.setAttribute('src', 'images/placeholder-image-square.jpg');
+  }
+}
+
+urlInput.addEventListener('input', updatePhoto);
+
+var form = document.querySelector('#form');
+var title = document.querySelector('#title-text');
+var notes = document.querySelector('#notes-text');
+var savedJournal = [];
+var entryID = 0;
+
+function submitListener(event) {
+  event.preventDefault();
+
+  savedJournal.unshift({
+    title: title.value,
+    photoURL: urlInput.value,
+    notes: notes.value,
+    entryID: entryID
+  });
+  entryID++;
+  var jsonJournal = JSON.stringify(savedJournal);
+  localStorage.setItem('journalEntry', jsonJournal);
```

Comment:
All `localStorage` related code should be in `data.js` so that it is in the same file as the data it is preserving.

Reply:
Also, there is no need to save data in localStorage after every submitted entry, you just need to do it before the page unloads. I encourage you to look back at `javascript-local-storage`, the code you'll need for this project will be very similar to the code you wrote in that exercise.